packages/mobile/src/screens/favorites-screen/PlaylistsTab.tsx

```diff
@@ -35,6 +36,16 @@ export const PlaylistsTab = () => {
   const isOfflineModeEnabled = useIsOfflineModeEnabled()
   const isReachable = useSelector(getIsReachable)
   const isDoneLoadingFromDisk = useSelector(getIsDoneLoadingFromDisk)
+  const offlineDownloadStatus = useProxySelector(
+    (state: AppState) => {
+      if (isDoneLoadingFromDisk && isOfflineModeEnabled && !isReachable) {
+        return getOfflineDownloadStatus(state)
```

Comment:
I'm using `offlineDownloadStatus` as the source of truth, but it seems to be very error prone right now... like it doesn't always match up with what is ultimately shown on the collection screen, which ultimately uses `offlineTracks`. So maybe it makes sense more sense to check `offlineTracks` instead of `offlineDownloadStatus` to be safe? On the other hand, if `offlineDownloadStatus` is indeed always supposed to match `offlineTracks` (i.e. set of tracks with status SUCCESS equals set of tracks in `offlineTracks`), it's fine to keep it this way. Wdyt?

Reply:
@amendelsohn i think you might have some ideas here given our discussion about source of truth between cached tracks and offlineTracks?

Reply:
ooh interested (also add offlineDownloadStatus to that list if we're just talking source of truth for whether a track should be shown when offline or not)